.ci/test.sh

```diff
@@ -13,6 +13,14 @@ if [[ "${TASK}" == "r-package" ]]; then
     exit 0
 fi
 
+if [[ "$TASK" == "cpp-tests" ]]; then
+    mkdir $BUILD_DIRECTORY/build && cd $BUILD_DIRECTORY/build
+    cmake -DBUILD_CPP_TEST=ON -DUSE_OPENMP=OFF ..
```

Comment:
Why with `USE_OPENMP=OFF`?
Because OpenMP is not fully supported yet due to linking problems (https://github.com/microsoft/LightGBM/pull/3555#discussion_r579187716) and to prevent compatibility problems with sanitizers:
> // ... Note that OpenMP needs to be
    // disabled, otherwise thread sanitizer will also report false
    // positives.
 https://github.com/dmlc/xgboost/blob/f6fe15d11fd05ff1cfe77441fa6c5d7280ee2131/tests/cpp/test_learner.cc#L196-L199

Reply:
I think that's ok! Even if we're disabled openmp, this test is still an incremental improvement in test coverage that will catch other issues.